src/jit/gentree.cpp

```diff
@@ -374,11 +376,13 @@ void GenTree::InitNodeSize()
     static_assert_no_msg(sizeof(GenTreeLabel)        <= TREE_NODE_SZ_SMALL);
     static_assert_no_msg(sizeof(GenTreePhiArg)       <= TREE_NODE_SZ_SMALL);
     static_assert_no_msg(sizeof(GenTreeAllocObj)     <= TREE_NODE_SZ_LARGE); // *** large node
-#ifndef FEATURE_UNIX_AMD64_STRUCT_PASSING
+#ifndef FEATURE_PUT_STRUCT_ARG_STK
     static_assert_no_msg(sizeof(GenTreePutArgStk)    <= TREE_NODE_SZ_SMALL);
-#else  // FEATURE_UNIX_AMD64_STRUCT_PASSING
+#else  // FEATURE_PUT_STRUCT_ARG_STK
+    // TODO-Throughput: This should not need to be a large node. The object info should be
+    // obtained from the child node.
     static_assert_no_msg(sizeof(GenTreePutArgStk)    <= TREE_NODE_SZ_LARGE);
```

Comment:
Shouldn't we add an assert that size of GenTreeFieldList <= TREE_NODE_SZ_SMALL ?